google/auth/_default.py

```diff
@@ -58,14 +58,18 @@ def _load_credentials_from_file(filename):
 
     Raises:
         google.auth.exceptions.DefaultCredentialsError: if the file is in the
-            wrong format.
+            wrong format or is missing.
     """
-    with io.open(filename, 'r') as file_obj:
-        try:
-            info = json.load(file_obj)
-        except ValueError as exc:
-            raise exceptions.DefaultCredentialsError(
-                'File {} is not a valid json file.'.format(filename), exc)
+    try:
+        with io.open(filename, 'r') as file_obj:
+            try:
+                info = json.load(file_obj)
+            except ValueError as exc:
+                raise exceptions.DefaultCredentialsError(
+                    'File {} is not a valid json file.'.format(filename), exc)
+    except IOError as exc:
+        raise exceptions.DefaultCredentialsError(
+            'File {} was not found.'.format(filename), exc)
```

Comment:
@jonparrott Not sure if you'd prefer EAFP here or LBYL via `os.path.exists`. I don't think EAFP is **needed**, i.e. we don't expect someone to move the file while this is running, i.e. is `os.path.exists()` is `True`, then we expect the file will still be there when we `io.open()` it.

I say this because the EAFP approach makes a bit more nesting than is fun for a developer's eyes.